### PR TITLE
Adds socket authentication and JWT fetching

### PR DESCRIPTION
**Pull request by ashfurrow**

### CODE REVIEW

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Warning">
        
            2 Warnings
        
      </th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">Needs testing on a Phone if change is non-trivial</td>
    </tr>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">Detected some Swift building time outliers</td>
    </tr>
    
    
  </tbody>
</table>
  

  



Time | Class | Function |
| --- | ----- | ----- |257.2ms | [AuctionViewController.swift](/artsy/eigen/tree/socket-events/Artsy/View_Controllers/Auction/AuctionViewController.swift#L176) | @objc dynamic func showRefineTappedAnimated(animated: Bool)
169.4ms | [WebSocket.swift](/artsy/eigen/tree/socket-events/Pods/Starscream/Source/WebSocket.swift#L528) | private func processRawMessage(buffer: UnsafePointer<UInt8>, bufferLen: Int)
57.6ms | [RefinementOptionsViewController+Private.swift](/artsy/eigen/tree/socket-events/Artsy/View_Controllers/Auction/RefinementOptionsViewController+Private.swift#L63) | func stackView() -> ORStackView




<p align="right" data-meta="generated_by_danger" data-base-commit="" data-head-commit="" >
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

File: Artsy/Networking/Live_Auctions/LiveAuctionSocketCommunicator.swift [hunk 1]
```diff
@@ -16,27 +16,32 @@ protocol SocketType: class {
...
     private let socket: SocketType
     private let causalitySaleID: String
     private var timer: NSTimer? // Heartbeat to keep socket connection alive.
 
     let updatedAuctionState = Observable<AnyObject>()
+    let newEvents = Observable<AnyObject>()
 
-    convenience init(host: String, causalitySaleID: String, accessToken: String) {
-        self.init(host: host, accessToken: accessToken, causalitySaleID: causalitySaleID, socketCreator: LiveAuctionSocketCommunicator.defaultSocketCreator())
+    let jwt: String
```
Comment (orta) [inline Artsy/Networking/Live_Auctions/LiveAuctionSocketCommunicator.swift]: could get `typealias'd` to its own type

Comment (orta) [inline Artsy/Networking/Live_Auctions/LiveAuctionSocketCommunicator.swift]: ah, you do this below: `typealias JWT = String`

**Comment by orta** [comment]
👍

File: Artsy/Networking/Live_Auctions/LiveAuctionSocketCommunicator.swift (hunk 1 above)
Comment (ashfurrow) [inline Artsy/Networking/Live_Auctions/LiveAuctionSocketCommunicator.swift]: Yeah, this should be changed though. Good catch!

**Pull request merged**